taxonomy/utils.py

```diff
@@ -326,3 +333,47 @@ def translate_text(key, text, source_language, target_language):
         LOGGER.exception(message)
 
     return result
+
+
+def apply_batching_to_translate_large_text(course_key, source_text):
+    """
+    Apply batching if text to translate is large and then combine it again.
+
+    Arguments:
+        course_key (str): Key or id of the object to uniquely identify.
+        source_text (str): Text which needs to be translated.
+
+    Returns:
+        dict: Translated object which contains TranslatedText and SourceLanguageCode.
+    """
+    soup = BeautifulSoup(source_text, 'html.parser')
+    # Split input text into a list of sentences on the basis of html tags
+    sentences = soup.findAll()
+    translated_text = ''
+    source_text_chunk = ''
+    result = {}
+    source_language_code = ''
+    LOGGER.info(f'[TAXONOMY] Translate course description applying batching for key: {course_key}')
+
+    for sentence in sentences:
+        # Translate expects utf-8 encoded input to be no more than
+        # 5000 bytes, so we’ll split on the 5000th byte.
+
+        if len(sentence.encode('utf-8')) + len(source_text_chunk.encode('utf-8')) < AMAZON_TRANSLATION_ALLOWED_SIZE:
+            source_text_chunk = '%s%s' % (source_text_chunk, sentence)
```

Comment:
Can we add a test case to cover this line as well?

Reply:
Yes, I'm currently trying to figure out, how to add a test case for it.